general/README.md

```diff
@@ -208,15 +209,102 @@ Avoid variable indirection when possible. Reassigning variables when no transfor
 ```js
 // good
 _handleEvent = function(e) {
-    _doSomethingWithEvent(e.target.checked);   
+    _doSomethingWithEvent(e.target.checked);
 }
 
 // bad
 _handleEvent = function(e) {
     var checked = e.target.checked;
 
-    _doSomethingWithEvent(checked);   
+    _doSomethingWithEvent(checked);
 }
 ```
 
 **[⬆ back to top](#table-of-contents)**
+
+## Naming Conventions
+> There are only two hard things in Computer Science: cache invalidation and naming things. *Phil Karlton*
+
+### Variables: Classes or Newable Functions (meant to be factories).
+It should be singular and begin with a capital letter.
+
+```js
+var ticket = new Ticket(id: 555);
+```
+### Variables: helpers and commmon variables
+
+It should always start with a lower case and be camel Case if is composed by 2 words.
+
+```js
+    var pagination = require('pagination');
+
+    pages = pagination.getPages();
+```
+
+### Varialbles: boolean
+
+It sholud start either with `is` or `has`
+
+```js
+var isAvailable = true,
+    hasTickets = false;
+```
+### Variables: jquery element or elements
+
+It sholud start with a dollar sign (`$`) so we can identify quickly by static analysis (not running and instecting the actuall code) that we are dealing with a jquery element/s.
+
+```js
+var $elements = $('someSelector'), // in case a global search
+    $element = this.$('some selector'); // in case of backbone
+```
+### Variables: deferred
+
+we recomend using `dfd or promise` as part of a deferred name.
```

Comment:
If a variable contains a [`jQuery.Deferred()`](https://api.jquery.com/jquery.deferred/) or [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), the variable should end in `Dfd` or `Promise`, respectively: